src/main/java/org/mastodon/mamut/treesimilarity/NodeMappings.java

```diff
@@ -0,0 +1,130 @@
+package org.mastodon.mamut.treesimilarity;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+
+import org.mastodon.mamut.treesimilarity.tree.Tree;
+
+/**
+ * Utility class for {@link ZhangUnorderedTreeEditDistance} that provides
+ * static factory methods for the easy creation of {@link NodeMapping}s.
+ */
+class NodeMappings
+{
+	private NodeMappings()
+	{
+		// prevent from instantiation
+	}
+
+	/**
+	 * @return An empty {@link NodeMapping} with the specified cost.
+	 * (Please note that the costs for an empty mapping are almost never
+	 * zero!)
+	 */
+	public static < T > NodeMapping< T > empty( double cost )
+	{
+		return new EmptyNodeMapping<>( cost );
+	}
+
+	/**
+	 * @return A {@link NodeMapping} that represents a singleton Map from
+	 * tree1 to tree2 with the specified cost.
+	 */
+	public static < T > NodeMapping< T > singleton( double cost, Tree< T > tree1, Tree< T > tree2 )
+	{
+		return new SingletonNodeMapping<>( tree1, tree2, cost );
+	}
+
+	/**
+	 * @return A {@link NodeMapping} that represents a composed that contains
+	 * all the map entries of the given {@code children}. The costs of the
+	 * composed mapping is the sum of the costs of the children.
+	 */
+	@SafeVarargs
+	public static < T > NodeMapping< T > compose( NodeMapping< T >... children )
```

Comment:
I think, it would be more explicit, if ComposedNodeMapping was returned by this method.

Reply:
see https://github.com/mastodon-sc/mastodon-deep-lineage/pull/33#discussion_r1315855841